concurrency/runner_test.go

```diff
@@ -161,8 +157,45 @@ func TestForEach_ShouldBreakOnFirstError_ContextCancellationUnhandled(t *testing
 	assert.Equal(t, int32(2), processed.Load())
 }
 
-func TestForEach_ShouldReturnImmediatelyOnNoJobsProvided(t *testing.T) {
-	require.NoError(t, ForEach(context.Background(), nil, 2, func(ctx context.Context, job interface{}) error {
+func TestForEachJob_ShouldReturnImmediatelyOnNoJobsProvided(t *testing.T) {
+	var processed atomic.Int32
+	require.NoError(t, ForEachJob(context.Background(), 0, 2, func(ctx context.Context, idx int) error {
+		processed.Inc()
 		return nil
 	}))
+	require.Zero(t, processed.Load())
+}
+
+func BenchmarkForEachJob(b *testing.B) {
```

Comment:
We expect that individual jobs are long-running, so benchmarking iteration function seems to be of very low value. (To the point that I'd actually suggest removing the benchmark, so that we don't need to maintain it.)

Reply:
I agree about the very low value, removed in https://github.com/grafana/dskit/pull/113/commits/5b0be335c1cdccbb92e460a538ae7db17f203a75